### ISSUE TITLE
[Bug]: Externally connected domain's status is based on connection expiration instead of domain registrations expiration

### ISSUE THREAD

**Issue by villanovachile** [Automattic/wp-calypso#63633]
### Quick summary

In Upgrades > Domains, the status for an externally connected domains is based on the connection's expiration, rather than the domain's actual expiration. As a result, if the domain connection is expiring within 30 days, it will display "Expiring soon", regardless if the domain's expiration is beyond 30 days from now. This report is being filed based off a user's experience & confusion: #33443342-hc

Although this will be most noticeable to users on a monthly billing plan, there are still a number of users that are on the monthly subscription.

### Steps to reproduce

1. Add a domain connection to a plan set to expire within the next 30 days, for a domain that is set to expire beyond 30 days
2. Confirm that the domain connection is set to expire within 30 days, and ensure the domain is properly pointing to WPCOM
3. Go to Upgrades > Domains, and note the status displays "Expiring Soon", even if the domain itself is set to expire

It's import that the plan is set to expire within 30 days, because the domain connection's expiration is determined by the plan's expiration.

### What you expected to happen

I'd expect the status to either display that the domain connection is expiring, or to show the domain's registration expiration based on the WHOIS data. 

### What actually happened

Instead it shows "Expiring soon" with no clarification for whether the domain is expiring soon, or if the connection is expiring soon.

![image](https://user-images.githubusercontent.com/65082164/168449960-24b6cd1e-f340-4dba-8213-dc186d20b771.png)

### Context

customer report

### Browser

Google Chrome/Chromium, Mozilla Firefox, Microsoft Edge, Apple Safari, iOS Safari, Android Chrome

### Simple/Atomic

Simple, Atomic

### Other notes

We need to determine whether it is best to display a clarification in the domain's status that this pertains to the domain's connection, or whether we instead want the status to reflect the domain's registration status based on WHOIS data

### Reproducibility

Consistent

### Severity

Most (> 50%)

### Available workarounds?

No but the platform is still usable

**Comment by happychait** [comment]
It would be helpful if we could get some clarification on this part @villanovachile added in the description:

> Instead it shows "Expiring soon" **with no clarification for whether the domain is expiring soon, or if the connection is expiring soon.**

(emphasis mine)

**Comment by villanovachile** [comment]
After discussing this further with @delputnam we determined it is best to clarify the status better to instead display "Domain connection is expiring soon".

We also discussed that the message should not be displaying at all if auto-renew is enabled for the associated plan. After further testing, I determined that the conditions are indeed in place for mapped domains expiring within 7 days, but not for domains expiring within 30 days.

I've gone ahead and added the correct conditions for domains expiring within 30 days and have modified the status for expiring domain connections to display "Domain connection expiring soon".

PR made in #63692
